delegator/src/sign/sign.service.ts

```diff
@@ -0,0 +1,70 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common'
+import { Transaction, Prisma } from '@prisma/client'
+import { PrismaService } from '../prisma.service'
+import { SignDto } from './dto/sign.dto'
+import { DelegatorError } from './helper/errors'
+import { signTxByFeePayer, validateTransaction } from './helper/utils'
+
+@Injectable()
+export class SignService {
+  constructor(private prisma: PrismaService) {}
+
+  async create(data: SignDto) {
+    const transaction = await this.prisma.transaction.create({ data })
+    try {
+      validateTransaction(transaction)
+    } catch (e) {
+      const msg = `DelegatorError: ${e.name}`
+      throw new HttpException(msg, HttpStatus.BAD_REQUEST)
+    }
+    const signedRawTx = await signTxByFeePayer(transaction)
+    this.updateSignedRawTransaction(transaction.id, signedRawTx)
```

Comment:
I think there should be `await`.

```suggestion
    await this.updateSignedRawTransaction(transaction.id, signedRawTx)
```